app/templates/main/volunteerProfile.html

```diff
@@ -0,0 +1,117 @@
+<h6 align="center">{{volunteer.phoneNumber}}</h6>
+</div>
+<div class="btn-group">
+   <ul class="nav nav-tabs nav-fill mx-3 mb-3" id="tabs-tab" role="tablist">
+     <li class="nav-item" role="presentation">
+       <a class="nav-link active" href="#">Upcoming Events</a>
+     </li>
+     <li class="nav-item">
+       <a class="nav-link" aria-current="page" href="/search_student">Service Transcript</a>
+     </li>
+   </ul>
+ </div><br>
+ <div class="row">
+   <table class="table">
+     {% if upcomingEvents|length > 0 %}
+       <thead class="tableHeader">
+         <th scope="col"><h4><strong>Program</strong></h4>
+         <th scope="col"><h4><strong>Event Name</strong></h4>
+         <th scope="col"><h4><strong>Event Date</strong></h4>
+         <th scope="col"><h4><strong>Time</strong></h4>
+        <th scope="col"><h4><strong>Location</strong></h4>
+       </thead>
+       {% for event in upcomingEvents %}
+       <tr>
+         <td>{{event.singleProgram.programName if event.singleProgram else '--'}}</td>
+         <td>{{event.name}}</td>
+         <td>{{event.startDate.strftime('%m/%d/%Y')}}</td>
+         <td>{{event.timeStart.strftime('%-I:%M %p')}}</td>
+         <td>{{event.location}}</td>
+       </tr>
+       {% endfor %}
+     {% else %}
+       <h2><p class="text-info">Choose interests below to be notified about upcoming events.</p><h2>
+     {% endif %}
+   </table>
+ </div>
+ <br>
+ <br>
+ <br>
+ <!-- Start Program Elegibility Table -->
+ <div class="row">
+   <table class="table">
+            <tbody id="volunteerTable">
+              <thead>
+                <tr>
+                  <th scope="col">Program</th>
+                  <th scope="col">Interested?</th>
+                  <th scope="col">Completed Training</th>
+                  {% if user.isCeltsAdmin %}
+                  <th scope="col">Elegibility</th>
+                  {% endif %}
+                </tr>
+              </thead>
+              {% for program in eligibilityTable %}
+              <tr>
+                <td id="programNameTable">{{program.program.programName}}</td>
+                <td><input class="form-check-input" type="checkbox" id="{{program.program.id}}" name="{{volunteer.username}}" {% if program.program in interests_ids %} checked {% endif %}></td>
+                <td>{% if volunteer.username in trainingChecklist[program.program.id] %}<i class="bi bi-check-lg"></i> {% else %} <i class="bi bi-x-lg"></i>{% endif %}</td>
+                {% if user.isCeltsAdmin %}
+                <td class="programBan">
+                  {% if program.isNotBanned %}<i class="bi bi-check-lg"></i><button type="button" class="ban btn btn-link" id="{{program.program}}" name="{{program.program.programName}}" value="Ban">Edit</button>
+                  {% else %} <i class="bi bi-x-lg"></i><button type="button" class="ban btn btn-link" id="{{program.program}}" name="{{program.program.programName}}" value="Unban">Edit</button>

```

Comment:
even though `program.program` works here, you're relying on the object being automatically cast to a string that is just the id. I would be explicit and use `program.program.id`

Reply:
Fixed!